### PR TITLE
[14.0][MIG] mail_preview_base: Migration to 14.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 14.0 (In progress according to https://github.com/OCA/dms/pull/66)

Ping @etobella 

@Tecnativa TT28930

### CODE REVIEW

File: mail_preview_base/static/src/js/preview.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+ **********************************************************************************/
+
+odoo.define("mail_preview_base.preview", function (require) {
+    "use strict";
+
+    var DocumentViewer = require("mail.DocumentViewer");
+    var basic_fields = require("web.basic_fields");
+    var registry = require("web.field_registry");
+    var AttachmentBox = require("mail.AttachmentBox");
```
Comment (etobella) [inline mail_preview_base/static/src/js/preview.js]: This does not exist on 14.0 it has been replaced by https://github.com/odoo/odoo/blob/14.0/addons/mail/static/src/components/attachment_box/attachment_box.js

**Comment by etobella** [comment]
I tried to modify a binary field and use the new widget and it is not working. Can you check it please?

He says that the widget does not exist.

File: mail_preview_base/static/src/js/preview.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            attachmentViewer.appendTo($("body"));
+        },
+        _renderReadonly: function () {
+            this._super.apply(this, arguments);
+            if (this.value) {
+                this.attachment = {
+                    mimetype: this.recordData.res_mimetype,
+                    id: this.res_id,
+                    fileType: this.recordData.res_mimetype,
+                    name: this.filename,
```
Comment (chienandalu) [inline mail_preview_base/static/src/js/preview.js]:
```suggestion
                    mimetype: this.recordData.mimetype,
                    id: this.res_id,
                    fileType: this.recordData.mimetype,
                    name: this.recordData.name,
```

File: mail_preview_base/static/src/js/preview.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+                };
+                var mimetype = this.recordData.res_mimetype;
```
Comment (chienandalu) [inline mail_preview_base/static/src/js/preview.js]:
```suggestion
                var mimetype = this.recordData.mimetype;
```

**Review by chienandalu — CHANGES_REQUESTED** [review]
I think these changes will do it :wink:

**Review by chienandalu — APPROVED** [review]

**Comment by victoralmau** [comment]
Sorry, not ready yet (not work and need to add test if it's possible).

**Comment by victoralmau** [comment]
IMO `AttachmentBox` it's not necessary because it's default odoo functionality.

**Comment by pedrobaeza** [comment]
@etobella maybe you can help here :pray:

**Comment by chienandalu** [comment]
> Sorry, not ready yet (not work and need to add test if it's possible).

Doesn't it?
Peek 25-03-2021 17-58

> IMO AttachmentBox it's not necessary because it's default odoo functionality.

Looks like it. Anyway, the extensions like `mail_preview_audio` will have to deal with the new `owl` controller extending this function I'd say: https://github.com/odoo/odoo/blob/14.0/addons/mail/static/src/models/attachment/attachment.js#L253

**Comment by victoralmau** [comment]
> `mail_preview_audio`

Sorry but `mail_preview_audio` not migrated yet to 14.0

**Comment by etobella** [comment]
@victoralmau I made a small PR fixing this addon Tecnativa#3

**Comment by victoralmau** [comment]
> @victoralmau I made a small PR fixing this addon [Tecnativa#3](https://github.com/Tecnativa/social/pull/3)

Thanks a lot @etobella , i have applied your suggestion in these PR and added you as co-author.
I have testing it in https://github.com/OCA/dms/pull/66

File: mail_preview_base/views/ir_attachment_view.xml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+    <record id="view_attachment_form_mail_preview_base" model="ir.ui.view">
+        <field name="model">ir.attachment</field>
+        <field name="inherit_id" ref="base.view_attachment_form" />
+        <field name="priority">99</field>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='datas']" position="replace">
```
Comment (pedrobaeza) [inline mail_preview_base/views/ir_attachment_view.xml]: no need to replace, but only edit attributes and change widget.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**